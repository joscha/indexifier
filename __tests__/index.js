const fs = require('fs');
const path = require('path');
const indexifier = require('../');

const fixturesDir = path.join(__dirname, 'fixtures');

describe('indexifier', () => {
    describe('should be able to transform a directory structure', () => {
        const dir = path.join(fixturesDir, '1');
        it('as-is', () => {
            const ret = indexifier(dir);
            expect(ret).toMatchSnapshot();
        });

        describe('with filtering', () => {
            it('by extensions', () => {
                const ret = indexifier(dir, { fileTypes: ['.html'] });
                expect(ret).toMatchSnapshot();
            });

            it('by regexp', () => {
                const ret = indexifier(dir, { fileTypes: [], include: '^a.(.+)$' });
                expect(ret).toMatchSnapshot();
            });

            it('by extentions and regexp', () => {
                const ret = indexifier(dir, { fileTypes: ['.html'], include: '^a.(.+)$' });
                expect(ret).toMatchSnapshot();
            });
        });

        describe('html', () => {
            it('to HTML', () => {
                const ret = indexifier(dir, { isHtml: true });
                expect(ret).toMatchSnapshot();
            });

            it('w/o linking folders', () => {
                const ret = indexifier(dir, { isHtml: true, linkFolders: false });
                expect(ret).toMatchSnapshot();
            });
        });
    });
    it('can exclude files and folders', () => {
        const dir = path.join(fixturesDir, '2');
        const ret = indexifier(dir, {
            exclude: 'node_modules|a.txt',
        });
        expect(ret).toMatchSnapshot();
    });
    it('can ignore folders that do not contain files', () => {
        const dir = path.join(fixturesDir, '3');
        expect(indexifier(dir, { fileTypes: ['.html'] })).toMatchSnapshot();
        expect(indexifier(dir, { fileTypes: ['.html'], emptyDirectories: false })).toMatchSnapshot();
    });
});
