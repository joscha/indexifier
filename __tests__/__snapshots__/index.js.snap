// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`indexifier can exclude files and folders 1`] = `
"2
└── a.html
"
`;

exports[`indexifier can ignore folders that do not contain files 1`] = `
"3
├── a.html
└─┬ ignore
  └── me
"
`;

exports[`indexifier can ignore folders that do not contain files 2`] = `
"3
└── a.html
"
`;

exports[`indexifier should be able to transform a directory structure as-is 1`] = `
"1
├─┬ A
│ └── c.html
├── a.html
├── a.txt
└── b.html
"
`;

exports[`indexifier should be able to transform a directory structure html to HTML 1`] = `
"<!doctype html>
<html>
    <head>
        <meta charset=\\"UTF-8\\">
        <title>1</title>
        <style>
            * {
                font-family: monospace;
            }
        </style>
    </head>
    <body>
        <pre>
<a href=\\"./\\">1</a>
├─┬ <a href=\\"./A\\">A</a>
│ └── <a href=\\"./A/c.html\\">c.html</a>
├── <a href=\\"./a.html\\">a.html</a>
├── <a href=\\"./a.txt\\">a.txt</a>
└── <a href=\\"./b.html\\">b.html</a>
        </pre>
    </body>
</html>
"
`;

exports[`indexifier should be able to transform a directory structure html w/o linking folders 1`] = `
"<!doctype html>
<html>
    <head>
        <meta charset=\\"UTF-8\\">
        <title>1</title>
        <style>
            * {
                font-family: monospace;
            }
        </style>
    </head>
    <body>
        <pre>
1
├─┬ A
│ └── <a href=\\"./A/c.html\\">c.html</a>
├── <a href=\\"./a.html\\">a.html</a>
├── <a href=\\"./a.txt\\">a.txt</a>
└── <a href=\\"./b.html\\">b.html</a>
        </pre>
    </body>
</html>
"
`;

exports[`indexifier should be able to transform a directory structure limited to maxDepth of 1 1`] = `
"fixtures
├── 1
├── 2
└── 3

exports[`indexifier should be able to transform a directory structure with filtering by extensions 1`] = `
"1
├─┬ A
│ └── c.html
├── a.html
└── b.html
"
`;

exports[`indexifier should be able to transform a directory structure with filtering by extentions and regexp 1`] = `
"1
└── a.html
"
`;

exports[`indexifier should be able to transform a directory structure with filtering by regexp 1`] = `
"1
├── a.html
└── a.txt
"
`;
